### PR TITLE
Clean up operator changes listener example

### PR DESCRIPTION
**Pull request by deiwin**
Cleans up formatting and some logic errors in the example code. Uses
methods like `Array.prototype.filter` instead of for loops.

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -301,59 +301,57 @@ Notifies listener of operator state updates (availability, media)
...
 
 function onOperatorStatusChange(updatedOperator) {
-    var availableOperators = 0;
-    for (var i = 0; i < operators.length; i++) {
-        if (operators[i].id = updatedOperator.id) {
-            operators[i] = updatedOperator;
-            break;
-        }
+  operators = operators.map(function(operator) {
+    if operator.id == updatedOperator.id {
```
Comment (indrekj) [inline apiary.apib]: parantheses

Comment (deiwin) [inline apiary.apib]: Thanks

File: apiary.apib [hunk 2]
```diff
@@ -301,59 +301,57 @@ Notifies listener of operator state updates (availability, media)
 
 > + Sample usage: different behaviour depending on how many operators are available.
 
-```
-var useElement = false
-var operators = []
+```js
+var elementEnabled = false;
+var operators = [];
```
Comment (indrekj) [inline apiary.apib]: is this needed? All the methods already take `operators` as an argument, do we want to put them to a global object as well?

Comment (deiwin) [inline apiary.apib]: `#onOperatorStatusChange()` only gets that one changed operator, but needs to call `#updateElementUI()` with the entire list of operators. So we do need to hold state.

Comment (ostap0207) [inline apiary.apib]: Yeah, this is needed. In the case when just operator status changes we have to store other operators somewhere and update the one from them respectively.

File: apiary.apib [hunk 3]
```diff
@@ -301,59 +301,46 @@ Notifies listener of operator state updates (availability, media)
...
-        useElement = false;
-        // Start what to do when at least one operator becomes available
-
-        alert("At least one operator is now available.");
-
-        // End
-    }
+  var availableOperators = filterAvailableOperators(operators);
+  if (availableOperators.length == 0) {
+	alert("No operators are available.");
```
Comment (indrekj) [inline apiary.apib]: use 2 spaces like in the other examples

File: apiary.apib [hunk 4]
```diff
@@ -301,59 +301,46 @@ Notifies listener of operator state updates (availability, media)
...
 }
 
 function onOperatorListChange(updatedOperators) {
-    operators = updatedOperators;
-    updateElementUI(operators);
+  operators = updatedOperators;
+  updateElementUI(operators);
 }
 
 function updateElementUI(operators) {
```
Comment (indrekj) [inline apiary.apib]: I think it would make more sense if this function wouldn't take operators variable and it used the global instead

Comment (deiwin) [inline apiary.apib]: I guess

**Comment by deiwin** [comment]
Updated for all comments

File: apiary.apib (hunk 2 above)
Comment (urmastalimaa) [inline apiary.apib]: Doesn't that signal a shortcoming in our API?
Maybe pass all the operators as a second argument to the callback?

Comment (deiwin) [inline apiary.apib]: I agree. I feel the current use case will be a very common one, so we should make it as frictionless as possible. @ostap0207, do you know who were involved in the decision to not include the list in the `OPERATOR_STATUS_UPDATE` event?

**Comment by stenver** [comment]
+1

File: apiary.apib (hunk 2 above)
Comment (ostap0207) [inline apiary.apib]: There weren't any particular conversation to not include operator list. This came from the requirement https://salemove.atlassian.net/browse/RAD-123. It really depends on use cases. @justindipietro , @paniagua maybe you can provide more information. Having it as an `operator list` as an optional argument might indeed make it easier in some cases.

Comment (indrekj) [inline apiary.apib]: @deiwin Feel free to set up a meeting if you want to discuss this. I think we still have time to make changes if necessary.

Comment (deiwin) [inline apiary.apib]: @indrekj, as I'm currently working on planning some other upcoming changes for the JS API as well, I could call a meeting once we have a solid draft of the next version of the API, so we have more things to discuss.

Comment (deiwin) [inline apiary.apib]: I think this PR can be merged without these changes, though, as it reflects the current functionality of out API and improves the current live version of the docs.

**Comment by ostap0207** [comment]
!squash

File: apiary.apib (hunk 2 above)
Comment (deiwin) [inline apiary.apib]: Created a ticket: https://salemove.atlassian.net/browse/JSDK-1

**Pull request merged**